src/munge.ts

```diff
@@ -42,17 +42,71 @@ export function disableRemb(sdp: Sdp) {
 
 /**
  * Remove the codec with the given name (as well as any secondary codecs associated with
- * it) from the media blocks in the given SDP.
+ * it) from the media blocks in the given SDP or audio/video media description.
  *
- * @param sdp - The SDP from which to remove the given codec.
- * @param codecName - The name of the codec to filter.
+ * @param sdpOrAv - The {@link Sdp} or {@link AvMediaDescription} from which to remove the given codec.
+ * @param codecName - The name of the codec to remove.
  */
-export function removeCodec(sdp: Sdp, codecName: string) {
-  sdp.avMedia.forEach((media: AvMediaDescription) => {
+export function removeCodec(sdpOrAv: Sdp | AvMediaDescription, codecName: string) {
+  const mediaDescriptions = sdpOrAv instanceof Sdp ? sdpOrAv.avMedia : [sdpOrAv];
+  mediaDescriptions.forEach((media: AvMediaDescription) => {
     const codecInfos = [...media.codecs.entries()].filter(
       ([, ci]) => ci.name?.toLowerCase() === codecName.toLowerCase()
     );
-
     codecInfos.forEach(([pt]) => media.removePt(pt));
   });
 }
+
+/**
+ * Filter out unwanted codecs from the given SDP or audio/video media description.
+ *
+ * Note: Done this way because of a feature not implemented in all browsers, currently missing in
+ * Firefox. Once that is added we can use `RTPSender.getCapabilities` and filter those to call
+ * with `RTCRtpTransceiver.setCodecPreferences` instead of doing this manually.
+ *
+ * @param sdpOrAv - The {@link Sdp} or {@link AvMediaDescription} from which to filter codecs.
+ * @param allowedCodecNames - The names of the codecs that should remain in the SDP.
+ */
+export function filterCodecs(
```

Comment:
Wondering if we can name this something better...I feel like there's ambiguity around if the passed list are the ones being kept or the ones being removed.

Reply:
`retainCodecs` maybe

Reply:
Sure, I can also rename `filterCandidates` to `retainCandidates`.